keen/keen-dataviz.js
# Support for large datasets

## What does this PR do? How does it affect users?

This adds support for rendering large datasets.

Major changes:
* Adds `.paginate` method to public API, `.visibilityThreshold` option to `chartOptions`.
* Adds a new demo page (`test/demo/pagination.html`) to play around with these new features.
* `render` function in `libraries/default.js`: If the new dataviz methods (`paginate` or `visibilityThreshold`) are set to true, `render` will call new extension functions to render charts. These extension functions live in `libraries/c3_extensions/`

### Line Charts and Bar Charts

This allows for large datasets to be rendered in chunks, which improves the performance of the page by reducing the number of SVG elements with mouse event handlers on the page, prevents charts from breaking, and provides a cleaner user interface.

![linegraph](https://cloud.githubusercontent.com/assets/638351/12826236/4bd43aca-cb37-11e5-9830-0249d225b406.gif)

### Stacked Bar and Area Charts

If `stacked` is `true` and `paginate` is `true and there more than 15 data points, stacked bar and area graphs render an 'Other' bar.
 
<img width="1010" alt="screen shot 2016-02-17 at 3 06 39 pm" src="https://cloud.githubusercontent.com/assets/638351/13126542/16c3ce1c-d588-11e5-9e14-5f8198c94b0b.png">


### Pie and Donut Charts

A `visibilityThreshold` option is available for users. If the user sets it, it'll throw all data points with a fractional value less than that threshold into an "Other" slice. Setting this option will reduce the chance we'll render a pie with so many slices that it slows down the page and becomes unreadable. This feature is based off of a feature on google charts:

![piegraphs](https://cloud.githubusercontent.com/assets/638351/13126492/db97c24e-d587-11e5-9415-556ebe707461.gif)

## How should this be tested?

Run `gulp` and go to `demo/index.html`. Look at the "Paginated Line Graph" example and navigate around.

Then go to `demo/pagination.html`. Change the `visibilityThreshold` attribute for the pie/donut charts, and make sure every data point under the visibility threshold you set goes into the 'Other' slice. Play around with the `paginate` function as well.

Fetch the branch and/or deploy to staging to test the following:
